TypeCobol/Compiler/Sql/Model/TableReference.cs

```diff
@@ -0,0 +1,40 @@
+using System.Collections.Generic;
+using TypeCobol.Compiler.CodeElements;
+
+namespace TypeCobol.Compiler.Sql.Model
+{
+    
+    public class SingleTableReference : SqlObject
+    {
+        public TableViewCorrelationName TableOrViewName { get; }
+        public CorrelationClause CorrelationClause {get;}
+
+        public SingleTableReference(TableViewCorrelationName tableOrViewName, CorrelationClause correlation)
+        {
+            this.CorrelationClause = correlation;
+            TableOrViewName = tableOrViewName;
+        }
+
+        protected override bool VisitSqlObject(ISqlVisitor visitor)
+        {
+            return visitor.Visit(this);
```

Comment:
No. You must take into account the fields/properties of the class when implementing the Visit method. Here, TableOrViewName and CorrelationClause must be visited as children